src/pages/Verification/Verification.tsx

```diff
@@ -0,0 +1,115 @@
+import { useState, useEffect } from "react";
+import { Typography, Container, Box, Button } from "@mui/material";
+import { MailOutline } from '@mui/icons-material';
+import { selectToken, selectUser } from "@/redux/userSlice";
+import { useSelector } from "react-redux";
+import { useNavigate } from "react-router-dom";
+import { RootState } from '@/redux/store';
+import { BASE_URL } from '@/config/default';
+import axios, { AxiosError, AxiosResponse } from "axios";
+import AlertMessage from "@/components/ui/AlertMessage";
+
+interface AuthResponse {
+  success: boolean;
+  data?: any;
+  error?: string;
+  message?: string;
+}
+
+export default function Verification() {
+  const token = useSelector((state: RootState) => selectToken(state));
+  const user = useSelector((state: RootState) => selectUser(state));
+  const navigate = useNavigate();
+  const [countdown, setCountdown] = useState(120);
+  const [resendDisabled, setResendDisabled] = useState(true);  
+  const [alert, setAlert] = useState<{ message: string; severity: "error" | "warning" | "info" | "success" } | null>(null);
+
+  useEffect(() => {
+    if (!token) {
+      navigate('/DAOMatcher/login');
+    }
+    return () => {
+      clearInterval(countdown);
+    };
+  }, [token, navigate]);
+
+  useEffect(() => {
+    let countdownTimer: NodeJS.Timeout;
+    if (resendDisabled && countdown > 0) {
+      countdownTimer = setInterval(() => {
+        setCountdown(prevCountdown => Math.max(0, prevCountdown - 1));
+      }, 1000);
+    } else if (countdown === 0) {
+      setResendDisabled(false);
+    }
+    return () => clearInterval(countdownTimer);
+  }, [resendDisabled, countdown]);
+
+  const handleResendClick = async () => {
+    if (!resendDisabled) {
+      setResendDisabled(true);
+      setCountdown(120);
+
+      try {
+        const { data: successData }: AxiosResponse<AuthResponse> = await axios.get(
+          `${BASE_URL}/api/auth/confirm/resend`,
+          { withCredentials: true, headers: { Authorization: `Bearer ${token}` }
+           }
+        );
+
+        if (successData.success) {
+          setAlert({ message: successData.message || 'Verification email resent successfully', severity: 'success' });
+        } else {
+          setAlert({ message: successData.message || 'The email is already verified', severity: 'info' });
+        }
+      } catch (error) {
+        let errorData: AuthResponse | null = null;
+
+        if (error instanceof AxiosError) {
+          if (error.response && error.response.data) {
+            errorData = error.response.data;
+          }
+        }
+
+        const data = errorData ?? {
+          success: false,
+          data: null,
+          error: "Failed to resend verification email due to server error",
+          message: null,
+        };
+
+        setAlert({ message: `${data.error}`, severity: 'error' });
+      }
+
+      setTimeout(() => {
+        setResendDisabled(false);
+      },120000);
```

Comment:
How did you come up with time interval?

Reply:
Most email verifications part of a website have them. It is to give a chance for the user to check his email and also to not make him send emails repeatedly (I think).

Reply:
Okay cool. I like the idea :)